I see one of the tests fails, but it looks unrelated:

```
CC ../../ports/stm32/pin.c
../../ports/stm32/pin.c: In function 'pin_obj_init_helper':
../../ports/stm32/pin.c:379:32: error: 'GPIO_SPEED_FREQ_HIGH' undeclared (first use in this function)
     GPIO_InitStructure.Speed = GPIO_SPEED_FREQ_HIGH;
                                ^
../../ports/stm32/pin.c:379:32: note: each undeclared identifier is reported only once for each function it appears in
make: *** [build/ports/stm32/pin.o] Error 1
make: Leaving directory `/home/travis/build/micropython/micropython/ports/teensy'
The command "make -C ports/teensy" exited with 2.
```

--- a/py/nlrthumb.c
+++ b/py/nlrthumb.c
@@ -76,9 +76,9 @@ __attribute__((naked)) unsigned int nlr_push(nlr_buf_t *nlr) {
 #endif
     );
 
-    #if defined(__GNUC__)
+    #if defined(__GNUC__) && (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 8))
     // Older versions of gcc give an error when naked functions don't return a value
-    __builtin_unreachable();
+    return 0;
     #endif
 }
 

